internal/file/file.go

```diff
@@ -214,11 +223,14 @@ func CopyFileWithPerm(ctx context.Context, src, dst string, perm fs.FileMode) (n
 
 	exist, fi, err := exists(src)
 	switch {
-	case !exist || fi == nil || fi.Size() == 0:
+	case !exist, fi == nil, fi.Size() == 0, fi.IsDir():
 		return 0, errors.Wrap(err, errors.ErrFileNotFound(src).Error())
-	case err != nil && !os.IsExist(err):
+	case err != nil && (!errors.Is(err, fs.ErrExist) || errors.Is(err, fs.ErrNotExist)):
 		return 0, errors.Wrap(errors.ErrFileNotFound(src), err.Error())
+	case err != nil:
+		log.Warn(err)
 	}
```

Comment:
Is there any reason we use `switch` here?
I think we should check the `err` before checking `exist`?

Reply:
you prefer if branch?

Reply:
I think check order is not important in this case.

Reply:
I think both `switch` or `if` branch is fine, but seems using `if` here will reduce the complexity of the checking.

```go
if err != nil {
   log.Warn(err) // should we put before checking the error?
  if (!errors.Is(err, fs.ErrExist) || errors.Is(err, fs.ErrNotExist)) {
      return 0, errors.Wrap(errors.ErrFileNotFound(src), err.Error())
   }
}

if !exist, fi == nil || fi.Size() == 0 || fi.IsDir() {
   return 0, errors.Wrap(err, errors.ErrFileNotFound(src).Error())
}